static/quantum/photon_perf.js

```diff
@@ -0,0 +1,108 @@
+/* global fetch */
+import { curveLinear } from 'd3';
+import MG from 'metrics-graphics';
+import PropTypes from 'prop-types';
+import React from 'react';
+import Telemetry from './telemetry-node';
+
+const evolutionize = async (evolutions) => {
+  // XXX: The data fetched from the API has all the data but it does not compose
+  // a real Evolution object. The logic below turns the evolutions array into
+  // a real array of Evolution objects.
+  await new Promise((resolve) => {
+    Telemetry.init(resolve);
+  });
+
+  const versions = [];
+  const realEvolutions = evolutions.map(({ evo, version }) => {
+    const realEvo = new Telemetry.Evolution(
+      evo.buckets, evo.data, evo.kind, evo.description, evo.measure);
+    versions.push(version);
+    return realEvo;
+  });
+  return [realEvolutions, versions];
+};
+
+
+const evoTime = (params, graphEl, evolutions, legendLabels) => {
+  const dateses = evolutions.map(evo => evo.dates());
+  let percentileLabel = ' - medians';
+  if (params.percentile !== 50) {
+    percentileLabel = ` - ${params.percentile}th percentile'`;
+  }
+  const yLabel = evolutions[0].description + percentileLabel;
+  const valueses = evolutions.map(evo => evo.percentiles(params.percentile));
+  const datas = dateses.map((dates, i) => dates.map((date, j) => {
+    return {
+      date: date,
+      value: valueses[i][j],
+    };
+  }));
+  MG.data_graphic({
+    data: datas,
+    chart_type: 'line',
+    full_width: true,
+    full_height: true,
+    top: 0,
+    right: 0,
+    bottom: 40,
+    left: 70,
+    min_y_from_data: true,
+    target: graphEl,
+    x_extended_ticks: true,
+    x_label: params.useSubmissionDate ? 'Submission Date' : 'Built Date',
+    y_label: yLabel,
+    xax_format: date => `${date.getMonth() + 1}-${date.getDate()}`,
+    yax_format: y => y,
+    transition_on_update: false,
+    interpolate: curveLinear,
+    legend: legendLabels,
+    legend_target: graphEl.querySelector('.graph-legend'),
+    aggregate_rollover: true,
+  });
+};
+
+const processData = async ({ evolutions, identifier, plot }) => {
+  const rootEl = document.getElementById(identifier);
+  const graphEl = rootEl.getElementsByClassName('graph')[0];
+  const [realEvolutions, versions] = await evolutionize(evolutions);
+  if (plot.evoVersions) {
+    rootEl.getElementsByClassName('graph-container')[0].classList.add('evo-graph-container');
+    evoTime({
+      percentile: 50,
+      useSubmissionDate: plot.useSubmissionDate,
+    }, graphEl, realEvolutions, versions);
+  }
+};
+
+const createGraphEls = id => (
```

Comment:
Check out reacts refs mechanism to get hold of elements. You would move this function and processData into render and use refs to get the DOM elements that you’ll eventually need after mounting.